src/web/src/app/space/detail/component.ts

```diff
@@ -0,0 +1,274 @@
+import { Component, OnInit, ViewChild, Inject} from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { MatPaginator } from '@angular/material/paginator';
+import { DOCUMENT } from '@angular/common';
+import { FormGroup, FormBuilder, Validators} from '@angular/forms';
+import { startWith, map, debounceTime, distinctUntilChanged, switchMap } from 'rxjs/operators';
+import { Observable } from 'rxjs';
+
+import { Space } from '../model';
+import { SpaceInvitation } from '../model';
+import { SpaceLink } from '../model';
+import { SpaceUser } from '../model';
+
+import { SpaceService } from '../service';
+import { UserService } from '../../user/service';
+import { UserFields } from '../../user/model';
+import { User } from '../../user/model';
+
+type SpaceInvitationDetail = SpaceInvitation & UserFields;
+type SpaceUserDetail = SpaceUser & UserFields;
+
+@Component({
+	selector: 'app-space-detail',
+	templateUrl: './component.html',
+	styleUrls: ['./component.css']
+})
+export class SpaceDetailComponent implements OnInit {
+
+	invitationForm: FormGroup;
+	isInvitationFormHidden: boolean = true;
+
+	linkForm: FormGroup;
+	isLinkFormHidden: boolean = true;
+
+	currentSpace: Space;
+	currentUserId: string;
+	currentSpaceId;
+
+	linksURL: string = "/space/link/";
+
+	invitations: SpaceInvitationDetail[] = [];
+	links: SpaceLink[] = [];
+	users: SpaceUserDetail[] = [];
+
+	pageSize = {
+		invitations: 7,
+		links: 7,
+		users: 10
+	};
+
+	invitationsTotal: number;
+	linksTotal: number;
+	usersTotal: number;
+
+	users$: Observable<User[]>;
+
+	@ViewChild('invitationsPaginator') invitationsPaginator: MatPaginator;
+	@ViewChild('linksPaginator') linksPaginator: MatPaginator;
+	@ViewChild('usersPaginator') usersPaginator: MatPaginator;
+
+	constructor(
+		private route: ActivatedRoute,
+		@Inject(DOCUMENT) private document: any,
+		private fb: FormBuilder,
+		private space: SpaceService,
+		private user: UserService,
+	) {
+		this._initInvitationForm();
+		this._initLinkForm();
+	}
+
+	ngOnInit() {
+		this.currentUserId = this.user.info.id;
+
+		const pageSizeStr = localStorage.getItem('spaceDetailPageSize');
+		const pageSize = JSON.parse(pageSizeStr);
+		if (pageSize?.invitations && pageSize?.links && pageSize?.users){
```

Comment:
Добавь пожалуйста пробел между круглой и фигурной скобкой.
```suggestion
		if (pageSize?.invitations && pageSize?.links && pageSize?.users) {
```

Reply:
Готово (везде так сделал).